darts/timeseries.py

```diff
@@ -3005,6 +3037,598 @@ def map(
 
+    def window_transform(
+        self,
+        window_transformations: Union[Dict, Sequence[Dict]],
+        treat_na: Optional[Union[str, Union[int, float]]] = None,
+        forecasting_safe: Optional[bool] = True,
+        target: Optional["TimeSeries"] = None,
+        keep_non_transformed: Optional[bool] = False,
+        store_window_transformation: Optional[bool] = False,
+        suppress_warnings: Optional[bool] = False,
+    ) -> "TimeSeries":
+        """
+        Applies a moving window transformation over this ``TimeSeries``.
+
+        Parameters
+        ----------
+        window_transformations
+            A dictionary or a list of dictionaries.
+            Each dictionary specifes a different window transform.
+
+            The dictionaries should contain the following keys:
+
+            :``"function"``: the name of one of the builtin transformation functions, or to a callable function
+                             that can be applied to the input series by using ``pandas.DataFrame.rolling object``.
+
+                             The two following options are available for built-in transformation functions:
+
+                             * Based on ``pandas.DataFrame.rolling`` windows, ``"function"`` should map to one of
+                               ``{"count", "sum", "mean", "median", "min", "max", "std", "var", "skew", "kurt",
+                               "corr", "cov", "quantile", "sem", "rank"}``.
+                             * Based on ``pandas.DataFrame.ewm`` (Exponentially-weighted window),
+                               ``"function"`` should be one of ``{"mean", "var", "std", "corr", "cov", "sum",
+                               "skew"}``, prefixed by ``"ewm_"``. For example, ``"function": "ewm_mean"``
+                               or ``"function": "ewm_sum"``.
+
+            :``"window"``: the size of the window (in number of time steps) to be used for the transformation.
+
+            :``"components"``: Optional. A string or list of strings specifying the components to which the
+                               transformation should be applied. If not specified, the transformation will be
+                               applied to all components.
+
+            All other dictionary items provided will be treated as keyword arguments for the function group
+            (i.e., ``pandas.DataFrame.rolling`` or ``pandas.DataFrame.ewm``) or for the specific function
+            in that group (i.e., ``pandas.DataFrame.rolling.mean/std/max/min...`` or
+            ``pandas.DataFrame.ewm.mean/std/sum``).
+            This allows for more flexibility in configuring how the window slides over the data, by providing for
+            example:
+
+            * ``"center"``: ``True``/``False`` to set the observation at the current timestep at the
+              center of the windows (default is False),
+            * ``"closed"``: ``"right"``/``"left"``/``"both"``/``"neither"`` to specify whether the right,
```

Comment:
true my doc mixed the two. 'closed' means where the interval is closed (hence whether the values at the borders of the interval are included or not). Since we allow the current time step to be included in the window, then this parameter will not be important anymore, we can actually leave it to default ('right') or specify that both ends are included ('both').

Reply:
I defaulted it to "both"